### PR TITLE
On-chain feature flags

### PR DESCRIPTION
**Pull request by mattkanwisher**

### CODE REVIEW

File: builtin/types/chainconfig/chainconfig.proto [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+message ListConfig {
+}
+
+message ListFeaturesRequest {
+}
+
+message ListFeaturesResponse {
+    repeated GetFeatureResponse features = 1;
```
Comment (enlight) [inline builtin/types/chainconfig/chainconfig.proto]: Don't nest response types, or request types. Create a `FeatureInfo` type instead:
```
message ListFeaturesResponse {
    repeated FeatureInfo features = 1;
}
```

File: builtin/types/chainconfig/chainconfig.proto [hunk 2]
```diff
@@ -0,0 +1,70 @@
+syntax = "proto3";
+
+import "github.com/loomnetwork/go-loom/types/types.proto";
+
+
+message ChainConfigInitRequest {
```
[the remark this answers is not in the document]
Comment (pathornteng) [reply builtin/types/chainconfig/chainconfig.proto]: Done

File: builtin/types/chainconfig/chainconfig.proto [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    string name = 1;
+    enum FeatureStatus {
+        PENDING = 0;
+        WAITING = 1;
+        ENABLED = 2;
+        DISABLED = 3;
+    }
+    FeatureStatus status = 2;
+    repeated Address validators = 3;
+    uint64 blockHeight = 4;
```
Comment (enlight) [inline builtin/types/chainconfig/chainconfig.proto]: -> `block_height`
And add a comment to describe what this is for

File: builtin/types/chainconfig/chainconfig.proto [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+message Params {
+    uint64 voteThreshold = 1;
+    uint64 numBlockConfirmations = 2;
```
Comment (enlight) [inline builtin/types/chainconfig/chainconfig.proto]: -> `vote_threshold`
-> `num_block_confirmations`

File: builtin/types/chainconfig/chainconfig.proto [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+message ListConfig {
+}
+
+message ListFeaturesRequest {
+}
+
+message ListFeaturesResponse {
+    repeated FeatureInfo featureInfos = 1;
```
Comment (enlight) [inline builtin/types/chainconfig/chainconfig.proto]: -> `features`

Comment (pathornteng) [reply builtin/types/chainconfig/chainconfig.proto]: We have both `Feature` and `FeatureInfo`. Won't this cause type confusion?

**Pull request merged**